### PR TITLE
Some minor fixes for LaTeX printer

### LINKED ISSUE #15934: LaTeX priniting: \mathrm vs. \text
It turns out that both `\mathrm` and `\text` is used for printing "text" in the LaTeX printing routines. It seems like there should be one that is standard and that it should be possible to change that by configuration.

### \text
 * **Pro** adapts the font to the used math-font, not necessarily a Roman font
 * **Con** Relies on the `amsmath` (or some other `ams*` package

### \mathrm
 * **Pro** works in default LaTeX without importing any packages
 * **Con** always gives a Roman (Serif) font, even if the current math font is not Roman

Some quick grep:ing tells us that there are 123 instances of `\mathrm` and 46 instances of `\text` in the .py-files.

Ideally, this should be configurable, but even then we need to decide on a standard. Personally, I will set it to `\text` the second it is implemented, but the package dependency may be confusing for newer users. (It seems like both options works in e.g. iPython but I have not explicitly tested, just not noted any issue.)

Probably the easiest way to implement it is to add a helper function `print_text` or something so people do not have to test at every single instance.

### PR DESCRIPTION
**Pull request by oscargus**
#### References to other Issues or PRs

Closes #15934

#### Brief description of what is fixed or changed
Some fixes for the LaTeX printer:
 * Always use `\text` instead of `\mathrm` (see #15934 )
 * Use `mathbf` instead of `\boldmath{\mathrm` (gives identical rendering)
 * Added tests for some DiffGeom printing
 * Made a unified function for printing of transforms
 * Corrected documentation for `imaginary_unit`

#### Other comments
`\text` is from `amsmath` but we rely on that heavily elsewhere. The "problem" with `\mathrm` is that `rm` stand for a Roman (Serif) font. In certain situations, the template use non-Roman fonts for the math and then `\text` works better as it simply gives the non-italic version of the math font.

#### Release Notes

* printing
  * `\text` is consistently used instead of `\mathrm` in LaTeX printer to get non-italic text.
  * Added option to select output format for `re` and `im` in LaTeX printer: `gothic_re_im`

### CODE REVIEW

File: sympy/printing/tests/test_latex.py [hunk 1]
```diff
@@ -1958,3 +1958,16 @@ def test_imaginary_unit():
     assert latex(1 + I, imaginary_unit='foo') == '1 + foo'
     assert latex(I, imaginary_unit="ti") == '\\text{i}'
     assert latex(I, imaginary_unit="tj") == '\\text{j}'
+
+
+def test_DiffGeomMethods():
+    from sympy.diffgeom import Manifold, Patch, CoordSystem, Point, BaseScalarField
+    r0, theta0 = symbols('r0, theta0')
```
Comment (oscargus) [inline sympy/printing/tests/test_latex.py]: These were apparently used later on in the code snippet I copied from... Will remove them, but waiting for review comments first.

File: sympy/printing/latex.py [hunk 2]
```diff
@@ -1494,7 +1494,7 @@ def _print_Transpose(self, expr):
 
     def _print_Trace(self, expr):
         mat = expr.arg
-        return r"\mathrm{tr}\left(%s \right)" % self._print(mat)
+        return r"\text{tr}\left(%s \right)" % self._print(mat)
```
Comment (asmeurer) [inline sympy/printing/latex.py]: This should be \operatorname

**Comment by asmeurer** [comment]
Are all the modified lines tested?

File: sympy/printing/latex.py (hunk 2 above)
Comment (oscargus) [reply sympy/printing/latex.py]: Makes sense.

**Comment by oscargus** [comment]
For the `format` ones I am not sure as the output doesn't change there I have not checked it up.

Looking at the `text` ones there are a few which doesn't seem to have a test. I added a few which I noted didn't have a test but other than that I more or less blindly replace `mathrm` with `text` in the tests. So didn't really reflect on it. It seems like `Differential` does not have a test currently.

**Comment by oscargus** [comment]
I've added a test for `Differential` (it turned out that is was tested as part of the `WedgeProduct` test, but not explicitly earlier.

For the other methods I have changed, I believe most are tested although again not explicitly created from that class. These are all Modules from the `polys.agca` part and there are basically no examples so I do not really know how to create them...

**Comment by oscargus** [comment]
I've made printing of `im`consistent with printing of `re`.

I also added an option to change the output between `\Re` and `\operatorname{re}` called `text_re_im`, which is `False` by default leading to the former output (which is what Sympy has used historically. @jksuom do you have opinions on the flag name?

**Comment by asmeurer** [comment]
How about `re_im_style` and it's a string instead of a boolean. text_re_im is misleading because it uses \operatorname, not \text. 

By the way, I vote to change the re/im style to by default be \operatorname. As I mentioned [here](https://github.com/sympy/sympy/pull/16065#issuecomment-468852020), the gothic style looks terrible, and it's rarely used as @jksuom noted (apparently it [originates from Knuth himself](https://tex.stackexchange.com/questions/192692/why-does-amsmath-use-fraktur-for-real-and-imaginary-parts)).

**Comment by oscargus** [comment]
What about `gothic_re_im` with Boolean? If ext-based, what are the options called? (I originally considered this but couldn't figure out the classifiers.) I agree that the current one may be a bit confusing, but the end result is text, although not `\text`...

**Comment by jksuom** [comment]
> I vote to change the re/im style to by default be \operatorname.

+1 

As to the keyword, `gothic_re_im` looks good to me. (There will probably be no need to use it explicitly if its default is false.)

**Comment by oscargus** [comment]
Changed to `gothic_re_im` with default `False`.

File: sympy/printing/latex.py [hunk 3]
```diff
@@ -2376,7 +2376,7 @@ def latex(expr, fold_frac_powers=False, fold_func_brackets=False,
         "ri" leads to ``\mathrm{i}`` which gives `\mathrm{i}`.
     gothic_re_im : boolean, optional
         If set to ``True``, `\Re` and `\Im` is used for ``re`` and ``im``, respectively.
-        The default is ``False`` leading to `\opeatorname{re}` and `\opeatorname{re}`.
+        The default is ``False`` leading to `\operatorname{re}` and `\operatorname{re}`.
```
Comment (jksuom) [inline sympy/printing/latex.py]: `re` and `im`?

Comment (oscargus) [reply sympy/printing/latex.py]: Yes, that is the next commit two minutes later...

**Comment by oscargus** [comment]
This is fixed and merged with current master.

**Pull request merged**